src/invalidations.jl

```diff
@@ -0,0 +1,180 @@
+export @snoopr, invalidation_trees, explain
+
+# This doesn't technically have to be mutable but it's more convenient for testing equality
+mutable struct InstanceTree
+    mi::MethodInstance
+    depth::Int32
```

Comment:
Do we need to use `Int32` here?

Reply:
Not strictly necessary but that's the type that gets loaded into the returned array (in C, `int == Int32`).